### PR TITLE
fix: customize-reservation-limit-date

### PR DESCRIPTION
**Pull request by kibidan5**

### CODE REVIEW

File: src/application/domain/experience/reservation/config.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+/**
+ * Get activity booking configuration from environment variable
+ * Environment variable format: {"activity-id-1":0,"activity-id-2":1,"activity-id-3":7}
+ */
+let configFromEnv: ActivityBookingConfig = {};
+try {
+    const envConfig = process.env.ACTIVITY_ADVANCE_BOOKING_DAYS_CONFIG;
+    if (envConfig) {
+        configFromEnv = JSON.parse(envConfig);
+        console.log('Loaded activity advance booking days config from environment variable');
```
[the remark this answers is not in the document]
Comment (kibidan5) [reply src/application/domain/experience/reservation/config.ts]: 確かにそうでした！
こちらで切り替えた https://github.com/Hopin-inc/civicship-api/pull/419/commits/edbd905ba308a69e33cb668bee4c7eba92aa8d3a

File: src/application/domain/experience/reservation/validator.ts [hunk 2]
```diff
@@ -55,10 +55,21 @@ export default class ReservationValidator {
...
+      if (now.getTime() >= slotStartAt.getTime()) {
+        throw new ReservationCancellationTimeoutError();
+      }
+      return;
+    }
+
+    // Otherwise, use the configured advance booking days
     const cancelLimit = new Date(slotStartAt);
-    cancelLimit.setDate(cancelLimit.getDate() - 1);
+    cancelLimit.setDate(cancelLimit.getDate() - advanceBookingDays);
```
[the remark this answers is not in the document]
Comment (709sakata) [reply src/application/domain/experience/reservation/validator.ts:72]: キャンセルは一律前日までということね？
それで良いと思います！

[the remark this answers is not in the document]
Comment (kibidan5) [reply src/application/domain/experience/reservation/validator.ts:72]: N日前まで申し込みでN日前までキャンセルはちょっと、キャンセル条件厳しいような気も冷静にしてきた
一律前日でもいいかもしれないと思ってきた。

File: src/__tests__/unit/experience/reservation.validator.test.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
 import "reflect-metadata";
-import { 
+import {
```
Comment (kibidan5) [inline src/__tests__/unit/experience/reservation.validator.test.ts:2]: > civicship-api@1.0.0 test
> jest --runInBand --testPathPattern=reservation.validator.test.ts

 PASS  src/__tests__/unit/experience/reservation.validator.test.ts (5.527 s)
  ReservationValidator
    validateReservable
      ✓ should pass when slot is valid and no conflicts and enough capacity (1 ms)
      ✓ should throw if slot is not scheduled (4 ms)
      ✓ should throw if slot has already started (1 ms)
      ✓ should throw if there are conflicting reservations
      ✓ should throw if participant count exceeds capacity
      ✓ should throw if booking is within the default advance booking period
      with custom advance booking days
        ✓ should pass if booking is within custom advance booking period for specific activity
        ✓ should throw if booking is within custom advance booking period for specific activity (1 ms)
        ✓ should allow booking until start time for activities with 0 advance booking days
    validateJoinable
      ✓ should pass and return availableParticipationId if joinable
      ✓ should throw if reservation is not accepted
      ✓ should throw if opportunitySlot has already started (1 ms)
      ✓ should throw if user already joined
      ✓ should throw if no available participations
    validateCancellable
      ✓ should pass if cancellation is before cancellation deadline (1 day before start)
      ✓ should throw if cancellation is within cancellation deadline (1 day before start)
      ✓ should handle exact boundary condition for cancellation deadline (1 day before start)
      ✓ should throw when exactly at cancellation deadline limit (1 day before start) (1 ms)
      ✓ should handle future daylight saving time transitions
      ✓ should handle future leap year edge cases
      ✓ should handle timezone edge cases with future dates
      ✓ should use the same cancellation deadline (1 day) for all activities regardless of their advance booking days

Test Suites: 1 passed, 1 total
Tests:       22 passed, 22 total
Snapshots:   0 total
Time:        5.663 s

File: src/application/domain/experience/reservation/validator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kibidan5) [reply src/application/domain/experience/reservation/validator.ts]: https://github.com/Hopin-inc/civicship-api/pull/419/commits/5c23ccc8be6dbd1223b8bea7c270f731aecd7223

で対応
